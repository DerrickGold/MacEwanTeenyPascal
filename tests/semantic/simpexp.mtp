(*Testing simple expression type checking*)
const test := 2;
var a, b : integer;

begin
	a := a + b - test;
	a := a div b * test - 2
end.