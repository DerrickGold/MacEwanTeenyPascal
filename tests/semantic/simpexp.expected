
Semantically Corrected AST
[Head] (Block, Block Statement) 
Symbol Table:============================================================================:
Identifiers    |Stack Offset  |Type                 |Value                           
-----------------------------------------------------------------------------------------
a              |0             |Int, Var             |                                
b              |4             |Int, Var             |                                
test           |NONE          |Int, Con             |2                               
=======================================================================Depth 0   :Size 8   
    [Child] (Const Section) 
        [Child] (Const Declaration) 
            [Child] 2, TOK_ID, test, (Literal) 
            [Child] 2, TOK_NUM, 2, (Literal) 
    [Child] (Variable Section) 
        [Child] (Variable Declaration, Variable List) 
            [Child] 3, TOK_ID, a, (Literal) 
            [Sibling] 3, TOK_ID, b, (Literal) 
            [Child] 3, TOK_KEY_INTEGER, integer, (Type, Integer) 
    [Child] (Statement List) 
        [Child] (Assign Statement) 
            [Child] 6, TOK_ID, a, (Variable) 
            [Child] 6, TOK_MINUS, -, (Simple Expression, Binary Adding Operator) 
                [Child] 6, TOK_PLUS, +, (Simple Expression, Binary Adding Operator) 
                    [Child] 6, TOK_ID, a, (Variable) 
                    [Child] 6, TOK_ID, b, (Variable) 
                [Child] 6, TOK_ID, test, (Constant, Integer) 
        [Sibling] (Assign Statement) 
            [Child] 7, TOK_ID, a, (Variable) 
            [Child] 7, TOK_MINUS, -, (Simple Expression, Binary Adding Operator) 
                [Child] 7, TOK_STAR, *, (Multiply Operator, Term) 
                    [Child] 7, TOK_KEY_DIV, div, (Multiply Operator, Term) 
                        [Child] 7, TOK_ID, a, (Variable) 
                        [Child] 7, TOK_ID, b, (Variable) 
                    [Child] 7, TOK_ID, test, (Constant, Integer) 
                [Child] 7, TOK_NUM, 2, (Constant, Integer) 
